### PR TITLE
fix(cmd/cli): Show single-mesh-enforced warning

### LINKED ISSUE #4438: "osm install" should show warning msg about single-mesh-enforcement preventing multiple installs
**Bug description**:

When an existing osm mesh is already installed with single mesh enforcement (default true), and “osm install” is run again, the cli should warn that installation cannot proceed because of single-mesh-enforcement.

As of now, the cli does not make this fact clear, and instead prevents installation by prompting the user to run the install command again with a new mesh name and mesh namespace. It should instead warn users that installation cannot proceed because of single mesh.

**Affected area (please mark with X where applicable)**:
- Install                    [X]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [X]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**: Appropriate prompt

**Steps to reproduce the bug (as precisely as possible)**:

- Install osm with single mesh enforcement (default true). Installing again with various options show incorrect prompt.

image

image

**How was OSM installed?**: See above.

**Anything else we need to know?**:

**Bug report archive**:

**Environment**:
- OSM version (use `osm version`): v1.0.0-rc.3
- Kubernetes version (use `kubectl version`):
- Size of cluster (number of worker nodes in the cluster):
- Others:

### PR DESCRIPTION
**Pull request by shalier**
**Description**:
Warns user that they cannot install a new mesh due to single-mesh-enforcement
and resolves #4438

Signed-off-by: Shalier Xia <shalierxia@microsoft.com>

**Testing done**:
Install osm with enforce-single-mesh true:
image
Install osm with enforce-single-mesh false:
image
image
image

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [x ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?no
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?no

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

**Review by snehachhabria — APPROVED** [review]
thanks for the fix!

File: cmd/cli/install_test.go [hunk 1]
```diff
@@ -351,7 +351,7 @@ var _ = Describe("Running the install command", func() {
 		})
 	})
 
-	Describe("when a mesh with the given name already exists", func() {
+	Describe("when a mesh with the given name already exists and enforceSingleMesh is false", func() {
```
Comment (snehachhabria) [inline cmd/cli/install_test.go:354]: It would be good if we can add tests for the scenarios you manually tested as well

**Review by johnsonshi — APPROVED** [review]

File: cmd/cli/install_test.go (hunk 1 above)
Comment (shalier) [reply cmd/cli/install_test.go:348]: There were multiple tests testing various cases of enforceSingleMesh, merged them into one test function

File: cmd/cli/install_test.go [hunk 2]
```diff
@@ -647,142 +643,79 @@ func TestEnforceSingleMesh(t *testing.T) {
...
-			fmt.Sprintf("osm.prometheus.retention.time=%s", testRetentionTime),
-			"osm.certificateProvider.serviceCertValidityDuration=24h",
-			"osm.deployPrometheus=true",
-			"osm.deployGrafana=false",
-			"osm.enableIngress=true",
-			"osm.certificateProvider.kind=tresor",
-		},
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fmt.Println(tc.name)
```
Comment (jaellio) [inline cmd/cli/install_test.go:701]:
```suggestion
```

**Review by snehachhabria — APPROVED** [review]
Thanks for cleaning up the tests too!

**Review by jaellio — APPROVED** [review]

**Pull request merged**